src/IK/src/DistanceTask.cpp

```diff
@@ -0,0 +1,243 @@
+                     errorPrefix,
+                     m_description);
+        return false;
+    }
+
+    // set the base frame name
+    if (!ptr->getParameter("reference_frame_name", m_referenceName))
+    {
+        log()->debug("{} [{}] No base_name specified. Using default \"\"",
+                     errorPrefix,
+                     m_description);
+    }
+
+    // set the finger tip frame Name
+    if (!ptr->getParameter("target_frame_name", m_targetFrameName))
+    {
+        log()->error("{} [{}] Failed to get the end effector frame name.",
+                     errorPrefix,
+                     m_description);
+        return false;
+    }
+
+    if (m_referenceName != "")
+    {
+        m_referenceFrameIndex = m_kinDyn->getFrameIndex(m_referenceName);
+
+        if (m_referenceFrameIndex == iDynTree::FRAME_INVALID_INDEX)
+        {
+            log()->error("{} [{}] The specified base name ({}) does not seem to exist.",
+                         errorPrefix,
+                         m_description,
+                         m_referenceName);
+            return false;
+        }
+    }
+
+    m_targetFrameIndex = m_kinDyn->getFrameIndex(m_targetFrameName);
+
+    if (m_targetFrameIndex == iDynTree::FRAME_INVALID_INDEX)
+    {
+        log()->error("{} [{}] The specified target name ({}) does not seem to exist.",
+                     errorPrefix,
+                     m_description,
+                     m_targetFrameName);
+        return false;
+    }
+
+    m_isInitialized = true;
+
+    return true;
+}
+
+bool DistanceTask::update()
+{
+    using namespace BipedalLocomotion::Conversions;
+    using namespace iDynTree;
+
+    m_isValid = false;
+
+    if (m_referenceName == "")
+    {
+        m_framePosition = toEigen(m_kinDyn->getWorldTransform(m_targetFrameIndex).getPosition());
+
+        if (!m_kinDyn->getFrameFreeFloatingJacobian(m_targetFrameIndex, m_jacobian))
+        {
+            log()->error("[DistanceTask::update] Unable to get the jacobian.");
+            return m_isValid;
+        }
+
+    } else
+    {
+        m_framePosition = toEigen(
+            m_kinDyn->getRelativeTransform(m_referenceFrameIndex, m_targetFrameIndex).getPosition());
+
+        if (!m_kinDyn->getRelativeJacobian(m_referenceFrameIndex,
+                                           m_targetFrameIndex,
+                                           m_relativeJacobian))
+        {
+            log()->error("[DistanceTask::update] Unable to get the relative jacobian.");
+            return m_isValid;
+        }
+
+        m_jacobian.rightCols(m_kinDyn->getNrOfDegreesOfFreedom()) = m_relativeJacobian;
+    }
+
+    m_computedDistance = sqrt(pow(m_framePosition(0), 2) + pow(m_framePosition(1), 2)
+                              + pow(m_framePosition(2), 2));
+
+    toEigen(this->subA(m_robotVelocityVariable))
+        = (m_framePosition.transpose() * m_jacobian.topRows<3>())
+          / (std::max(0.001, m_computedDistance));
```

Comment:
Can you remove 0.001 from the code? (I don't know if it makes sense to have it as parameter)

Reply:
It helps for the stability of the algorithm. I have moved it to an optional input parameter in https://github.com/ami-iit/bipedal-locomotion-framework/pull/717/commits/c47e5416e7393e9e2d7bb94cf93b688dbcd6f03a